src/shared_modules/utils/windowsHelper.h

```diff
@@ -175,6 +192,52 @@ namespace Utils
         return ret;
     }
 
+    static std::string normalizeTimestamp(const std::string& timestamp1, const std::string& timestamp2)
+    {
+        std::string normalizedTimestamp;
+
+        if (timestamp1.size() == INSTALLDATE_REGISTRY_VALUE_SIZE)
+        {
+            if (isNumber(timestamp1))
+            {
+                size_t pos = timestamp2.find(' ');
+
+                if (pos != std::string::npos)
+                {
+                    std::string date2_trimmed = timestamp2.substr(0, pos);
+                    std::string time2_trimmed = timestamp2.substr(pos + 1);
+
+                    Utils::replaceAll(date2_trimmed, "/", "");
+                    Utils::replaceAll(time2_trimmed, ":", "");
+
+                    if (date2_trimmed.size() == INSTALLDATE_REGISTRY_VALUE_SIZE
+                            || time2_trimmed.size() == HOURMINSEC_VALUE_SIZE)
+                    {
+                        if (date2_trimmed.compare(timestamp1) == 0)
+                        {
+                            normalizedTimestamp = timestamp2;
+                        }
+                        else
+                        {
+                            tm local_time_s {};
+                            local_time_s.tm_year = std::stoi(timestamp1.substr(0, 4)) - REFERENCE_YEAR;
+                            local_time_s.tm_mon = std::stoi(timestamp1.substr(4, 2)) - 1;
+                            local_time_s.tm_mday = std::stoi(timestamp1.substr(6, 2));
```

Comment:
where those exceptions are caught, is a danger if stoi throw an exception here?

Reply:
As far as I know, an exception is thrown if the argument is not a number or if it is not in the range of an int. At that point, I think the previous checks cover those cases. If an exception occurs for any other reason I don't have in mind it will let the return value as empty string.